### DISCUSSION TITLE
size of various Libs

### DISCUSSION THREAD

**Discussion by alokprasad** [DMTF/libspdm#603] [General]
In one of the presentations related to spdm , size of requester/responder were mentioned below

![image](https://user-images.githubusercontent.com/11716522/152910058-80482f2e-3566-4f04-aa78-ece9d1257f65.png)

Source(https://av.tib.eu/media/52509)

But recent development I see they are considerably large, 
around 500K with CLANG for reqiester/responder 
and consider libspdm_common_lib.a is common to both this is around 250K .

Can something be done to reduce size.
does this static library includes size with crypto lib included?

**Comment by jyao1** [comment]
This data is from a DLL after link. The LTO will strip unused functions. (thanks for MSVC)

.a might be larger, because it includes all function.

Here is latest data: 
![Untitled](https://user-images.githubusercontent.com/12147155/152911603-5a0a9197-2369-4201-91bb-bcc3a7d810fa.png)

**Comment by jyao1** [comment]
Some recent result on AARCH64, release version, full feature, with NULL crypto lib:

ARM-GCC compiler
```
-rwxr-xr-x 1 jyao1 jyao1 228016 Oct 29 21:52 bin/test_size_of_spdm_requester*
-rwxr-xr-x 1 jyao1 jyao1 225304 Oct 29 21:52 bin/test_size_of_spdm_responder*
```

ARM-CLANG compiler
```
-rw-r--r-- 1 jyao1 jyao1 43848 Oct 29 21:39 bin/test_size_of_spdm_requester.elf
-rw-r--r-- 1 jyao1 jyao1 54856 Oct 29 21:41 bin/test_size_of_spdm_responder.elf
```

**Comment by manojkiraneda** [comment]
@jyao1 do you the numbers for the latest released version x86 & arm numbers ? Raw, with and without crypto ?

**Reply by steven-bellock** [reply]
We do not really track sizes, since libspdm is so configurable. We basically wait for someone to file a size issue and then fix it.
